### PR TITLE
create the CanvasOnchain contract and write the test

### PR DESCRIPTION
**Pull request by Abidoyesimze**
## Description

_Concise description of proposed changes, We recommend using screenshots and videos for better description_

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #{issue number}_

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address:

### CODE REVIEW

File: packages/hardhat/contracts/CanvasOnchain.sol [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+     * @dev Reset a specific pixel (only owner)
+     * @param x X-coordinate of the pixel
+     * @param y Y-coordinate of the pixel
+     */
+    function resetPixel(uint256 x, uint256 y) external onlyOwner {
+        require(x < CANVAS_WIDTH, "X coordinate out of bounds");
+        require(y < CANVAS_HEIGHT, "Y coordinate out of bounds");
+        
+        delete canvas[x][y];
+    }
```
[the remark this answers is not in the document]
Comment (melanke) [reply packages/hardhat/contracts/CanvasOnchain.sol:122]: Well, we could override a pixel anyway, but, why erase? And without an event...

[the remark this answers is not in the document]
Comment (Abidoyesimze) [reply packages/hardhat/contracts/CanvasOnchain.sol:122]: Okay will make changes for that

[the remark this answers is not in the document]
Comment (Abidoyesimze) [reply packages/hardhat/contracts/CanvasOnchain.sol]: > Well, we could override a pixel anyway, but, why erase? And without an event...

I have remove the reset pixel functionality

**Review by melanke — APPROVED** [review]

**Review by melanke — CHANGES_REQUESTED** [review]
We need the deployment script, right?

**Comment by Abidoyesimze** [comment]
> We need the deployment script, right?

Ohhhh yes but i don't know if we are deploying it to optimism mainnet that's why i haven't deploy it yet

**Comment by Abidoyesimze** [comment]
> @Abidoyesimze Nice, a smart contract, thanks!
> 
> Please ensure you include a description for your PR. As a rule of thumb, always provide a proper description when creating a PR, it’s essential for collaboration.
> 
> Please give the branch always a dedicated name, what you want to implement. `smize` doesn't tell me anything and I think it was the same before for your personal page.
> 
> Sorry to stress that, but thats what we try to teach you here in the batch program. Pls fix this first, then I will have a look at your code.
> 
> And a deployment script would be nice. As soon as you have it deployed you can add the ABI to the `externalContract.ts` file in next.js. Then it is shown in the debug page.

Sure, i'll make changes to the branch name but are we deploying the contract to optimism mainnet

**Pull request closed without merging**